docs: CWC-288 Revise README.md (#178)
CoolBitX-Technology/coolwallet-sdk: README.md

### 2. Register and setup hardware wallet.

> To register you application with the wallet, take a look at the [wallet module](/packages/core/src/apdu/wallet)  in `core` package. This giud you through the process of registeration and seed generation.

Reviewer: giud ? 應該是錯字嗎 ？是不是 guide

Author: ok

Reviewer 2: This guides you through the process of registration and seed generation.

| --------------------------------------- | --------------------------------------------------------- | ------------------------ |
| [`@coolwallet/atom`](/packages/cws-atom) | ![version](https://img.shields.io/npm/v/@coolwallet/atom) | Cosmos |
| [`@coolwallet/bch`](/packages/cws-bch) | ![version](https://img.shields.io/npm/v/@coolwallet/bch) | Bitcoin Cash |
| [`@coolwallet/bnb`](/packages/cws-bnb) | ![version](https://img.shields.io/npm/v/@coolwallet/bnb) | Binance |
| [`@coolwallet/bsc`](/packages/cws-bsc) | ![version](https://img.shields.io/npm/v/@coolwallet/bsc) | Binance Smart Chain |
| [`@coolwallet/btc`](/packages/cws-btc) | ![version](https://img.shields.io/npm/v/@coolwallet/btc) | Bitcoin/USDT |
| [`@coolwallet/dot`](/packages/cws-dot) | ![version](https://img.shields.io/npm/v/@coolwallet/dot) | Polkadot/Kusama |
| [`@coolwallet/eth`](/packages/cws-eth) | ![version](https://img.shields.io/npm/v/@coolwallet/eth) | Ethereum (Ether, ERC20, Smart Contract, EIP-1559 etc.)|
| [`@coolwallet/icx`](/packages/cws-icx) | ![version](https://img.shields.io/npm/v/@coolwallet/icx) | Icon |

> | [`@coolwallet/ltc`](/packages/cws-ltc) | ![version](https://img.shields.io/npm/v/@coolwallet/ltc) | LetCoin |

Reviewer 2: ```suggestion
| [`@coolwallet/ltc`](/packages/cws-ltc) | ![version](https://img.shields.io/npm/v/@coolwallet/ltc) | Litecoin |
```

Author: ok

| [`@coolwallet/trx`](/packages/cws-trx) | ![version](https://img.shields.io/npm/v/@coolwallet/trx) | Tron |
| [`@coolwallet/xlm`](/packages/cws-xlm) | ![version](https://img.shields.io/npm/v/@coolwallet/xlm) | Stellar/Kinesis |
| [`@coolwallet/xrp`](/packages/cws-xrp) | ![version](https://img.shields.io/npm/v/@coolwallet/xrp) | Ripple |
| [`@coolwallet/zen`](/packages/cws-zen) | ![version](https://img.shields.io/npm/v/@coolwallet/zen) | Zen Cash |

## Examples: Build ETH in web app

> ### Create connect

Reviewer 2: ```suggestion
### To connect to CoolWallet Pro via BLE
```

Author: ok

```
npm install @coolwallet/core
npm install @coolwallet/transport-web-ble
```

```javascript
import WebBleTransport from "@coolwallet/transport-web-ble";
import * as core from "@coolwallet/core";
```

Create a connection to obtain the Card Name and SE Public Key.

```javascript

connect = async () => {
WebBleTransport.listen(async (error, device) => {
    const cardName = device.name;
    const transport = await WebBleTransport.connect(device);
    const SEPublicKey = await core.config.getSEPublicKey(transport)
    this.setState({ transport, cardName, SEPublicKey });
    localStorage.setItem('cardName', cardName)
    localStorage.setItem('SEPublicKey', SEPublicKey)
  });
};

disconnect = () => {
  WebBleTransport.disconnect(this.state.transport.device.id);
  this.setState({ transport: undefined, cardName: "" });
};

```

- transport: The object use to communicate with CoolWallet

> - SEPublicKey: The key use to check se.

Reviewer 2: ```suggestion
- SEPublicKey: The key used to authenticate SE.
```

Author: ok

Obtain app key pairs.

```javascript
const keyPair = crypto.key.generateKeyPair()
localStorage.setItem('appPublicKey', keyPair.publicKey)
localStorage.setItem('appPrivateKey', keyPair.privateKey)
```

> - keyPair: The keys use to check your app.

Reviewer 2: shall we move this part to next section?

Author: I will move this code to `Register application with CoolWallet Pro`

> ### Register card

Reviewer 2: ```suggestion
### Register application with CoolWallet Pro
```

Author: ok

Register card and obtain the appId.

```javascript
const name = 'your app name'
const SEPublicKey = localStorage.getItem('SEPublicKey')

> const appId = await apdu.pair.register(transport, appPublicKey, password, name, SEPublicKey);

Reviewer 2: shall we explain what `password` represent?

Reviewer 2: we could also leave note here to indicate a single CoolWallet Pro could only be paired to 3 devices.

Author: ok

```

### Create / Recover the wallet

Use function `setSeed` to create or recover your wallet.

```javascript
const seedHex = bip39.mnemonicToSeedSync(mnemonic).toString('hex');
await apdu.wallet.setSeed(transport, appId, appPrivateKey, seedHex, SEPublicKey)

```

If you want to create seed by card, you can use function `createSeedByCard`. And also choose the length of seed(12, 18, 24).

```javascript
await apdu.wallet.createSeedByCard(transport, appId, appPrivateKey, 12);
```

> ### Integration

Reviewer 2: ```suggestion
### Use coin app
```

Author: ok

  appPrivateKey,
  appId,
  addressIdx
); 

```

The address generated is compatible to BIP44 with **account** and **change** set to 0, which means calling `getAddress(i)` will get the address of folllowing BIP44 path:

```none
m/44'/60'/0'/0/{i}
```

In the design of current hardware, we only support path `m/44'/60'/0'/0/{i}` for speed optimization. This might change in the future and we will then open a more general interface to deal with custom path.

If you have `accountPublicKey` and `accountChainCode`, you can use the function `ETH.getAddressByAccountKey()` to get the address. 
```javascript
const address = await ETH.getAddressByAccountKey(
  accountPublicKey,
  accountChainCode,
  addressIndex
);
```

### Sign Transaction

The signedTx is signed by CoolWallet, which can be sent directly.

```javascript
const transaction = {
    nonce: "0x21d",
    gasPrice: "0x59682f00",
    gasLimit: "0x5208",
    to: "0x81bb32e4A7e4d0500d11A52F3a5F60c9A6Ef126C",
    value: "0x5af3107a4000",
    data: "0x00",
    chainId: 1
};
const signTxData = {
  transport,
  appPrivateKey,
  appId,
  transaction,
  addressIndex,
};

const signedTx = await ETH.signTransaction(signTxData);

```

> ## Integration

Reviewer 2: ```suggestion
## Contributing
```

Author: ok

> If you want to integrate the other coin into CoolWallet, you can refer to [CONTRIBUTING](./CONTRIBUTING.md) or contact us.

Reviewer 2: If you're interested to develop new coin for CoolWallet Pro, please see [CONTRIBUTING](./CONTRIBUTING.md) for more information.

const appId = await apdu.pair.register(transport, appPublicKey, password, name, SEPublicKey);
```

> - password: Need to set the password, which will be used to change app.

Reviewer 2: ```suggestion
- password: Pairing password for the app to establish the connection with CoolWallet Pro. The password could be supplied by the user.
```

Reviewer 2: Does the password have a max length limit?
